substrate/primitives/consensus/beefy/src/test_utils.rs

```diff
@@ -36,52 +48,161 @@ pub enum Keyring {
 	One,
 	Two,
 }
+/// Trait representing BEEFY specific generating and signing behavoir of authority id
+///
```

Comment:
nit: why are the `BeefySignerAuthority` and `GenericKeyring` introduced in `test_utils` - they are intended (and used within the keystore) as core functionality, no? Or is the plan to migrate them elsewhere once we deem the implementation ready for production? I lean towards putting them within their final destination already since they're gated by the `bls_experimental` flag anyway.

Reply:
I could. I wasn't aware keyring used only for test purposes could live there. I'll move it there.

I thought   `_Marker(PhantomData<AuthorityId>)` is ugly and confusing to as all other options are accounts except for this one. but If you prefer that I'd proceed as suh.

Reply:
> Can't we put these Keyring related things in the sp-keyring crate together with the others?

I don't think it is a good id.  Keyring here depends on bunch of traits such as `AuthorityIdBound` which is defined in  beefy/primitives.It also uses other traits defined in `test_utils` like `BeefySignerAuthority` which is both used in the keyring and other tests in test_utils. By moving BEEFY `Keyring`  to `keyring` trait where we are going to introduce a cyclic dependancy of `primitive::beefy::test_utils` depends on `keyring::beefy` which depends on `primitive::beefy` and that won't work.